python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -636,11 +636,66 @@ def get_root_unreconciled_ancestors(
 
         return root_unreconciled_ancestors
 
-    def is_reconciled(self, asset_partition: AssetKeyPartitionKey) -> bool:
-        """Returns a boolean representing if the given `asset_partition` is currently reconciled.
-        An asset (partition) is considered unreconciled if any of:
-        - It has never been materialized
-        - One of its parents has been updated more recently than it has
-        - One of its parents is unreconciled.
-        """
-        return len(self.get_root_unreconciled_ancestors(asset_partition=asset_partition)) == 0
+    @cached_method
+    def get_ancestors_missing_or_with_unreconciled_children(
```

Comment:
similar logic to the above method but gets the parents that changed instead of the unreconciled assets